src/main/java/net/fabricmc/mappingio/tree/MappingTree.java

```diff
@@ -79,13 +71,23 @@ default FieldMapping getField(String srcOwnerName, String srcName, String srcDes
 		return owner != null ? owner.getField(srcName, srcDesc) : null;
 	}
 
+	default FieldMapping getField(String ownerName, String name, String desc, int namespace) {
+		ClassMapping owner = getClass(ownerName, namespace);
+
+		return owner != null ? owner.getField(name, desc, namespace) : null;
+	}
+
 	default MethodMapping getMethod(String srcOwnerName, String srcName, String srcDesc) {
 		ClassMapping owner = getClass(srcOwnerName);
 
 		return owner != null ? owner.getMethod(srcName, srcDesc) : null;
 	}
 
-	void accept(MappingVisitor visitor) throws IOException;
+	default MethodMapping getMethod(String ownerName, String name, String desc, int namespace) {
+		ClassMapping owner = getClass(ownerName, namespace);
+
+		return owner != null ? owner.getMethod(name, desc, namespace) : null;
+	}
```

Comment:
these should be implementable on the view as well, this class can then call super + cast to the mutable type

Reply:
Won't this introduce an unsafe cast though? Since we are casting an object to its subtype.

Reply:
Normally yes, but in this case it should be fine to assume that the implementation yields the mutable type every time. The override doesn't exist in a vacuum and consistent behavior of the actual impl class is to be expected.